### PR TITLE
Batch tx

### PR DESCRIPTION
**Pull request by dorin-iancu**
Split the contract into 5 modules:
- The main module - functions specific for the bridge use-cases.
- The setup module - setup functions, mostly one-time setup or functions like whitelisting a token, etc.
- The multisig_general module - generic multisig actions that aren't necessarily specific for the purpose of the bridge, like proposing adding more quorum members, signing an action etc.
- The storage module - all #[storage] functions.
- The util module - mostly generic #[view] functions and some require_some_condition functions.

Additionally, both types of transactions (Elrond -> Ethereum and vice-versa) support batch transactions.

### CODE REVIEW

File: esdt-safe/src/lib.rs [hunk 1]
```diff
@@ -182,7 +194,7 @@ pub trait EsdtSafe {
         /* TODO: Uncomment once the integration is complete
         self.ethereum_fee_prepay_proxy(self.fee_estimator_contract_address().get())
```
Comment (sasurobert) [inline esdt-safe/src/lib.rs:195]: the gwei / eth price contract is ready and updated - please try if it works with that.

File: multisig/src/setup.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            "slash amount must be less than or equal to required stake"
+        );
+        self.required_stake_amount().set(&required_stake);
+        self.slash_amount().set(&slash_amount);
+
+        Ok(())
+    }
+
+    #[endpoint(deployChildContracts)]
+    fn deploy_child_contracts(
```
Comment (sasurobert) [inline multisig/src/setup.rs:45]: does all these code deploys work with 1.5billion gas ? :)

File: egld-esdt-swap/src/lib.rs [hunk 3]
```diff
@@ -24,24 +24,20 @@ pub trait EgldEsdtSwap {
...
         let wrapped_egld_token_id = self.wrapped_egld_token_id().get();
-        
+
         self.require_local_roles_set(&wrapped_egld_token_id)?;
-        self.send().esdt_local_mint(
-            self.blockchain().get_gas_left(),
-            wrapped_egld_token_id.as_esdt_identifier(),
-            &payment,
-        );
+        self.send()
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply egld-esdt-swap/src/lib.rs:29]: The one from examples does all the issue-ing and setting roles from the contract, this version is more "light", as it expects thr token to be already issued and the SC having its local roles set by the issue-ing account.

File: esdt-safe/src/lib.rs [hunk 4]
```diff
@@ -6,20 +6,26 @@ use transaction::*;
 
 elrond_wasm::imports!();
 
+const MAX_TX_BATCH_SIZE: usize = 10;
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply esdt-safe/src/lib.rs:9]: Will do.

File: esdt-safe/src/lib.rs (hunk 1 above)
Comment (dorin-iancu) [reply esdt-safe/src/lib.rs:195]: Will try to get the integration ready tomorrow.

File: multisig/src/setup.rs (hunk 2 above)
Comment (dorin-iancu) [reply multisig/src/setup.rs:45]: Works with 100 million.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**